packages/tracing/src/transaction.ts

```diff
@@ -76,7 +79,7 @@ export class Transaction extends SpanClass implements TransactionInterface {
    * @hidden
    */
   public setMetadata(newMetadata: TransactionMetadata): void {
-    this._metadata = { ...this._metadata, ...newMetadata };
+    this.metadata = { ...this.metadata, ...newMetadata };
```

Comment:
`setMetadata` suggests to me that the new metadata is going to be set (final metadata is the new metadata), but this is appending the new metadata to the "old" one (final metadata is old metadata _+ new metadata_); so I'd suggest renaming the method. If the desired action is to `set`, then don't append; if we want to append, rename to eg `appendMetadata`.

Reply:
I'd agree if not for the fact, that we'd then have to update the behavior of `setExtras` and `setTags`, which work exactly like the implementation above.

Reply:
What Kamil said. Honestly, I'd get rid of the method all together now that the property is public if it wouldn't be a breaking change.